src/app/pages/profile/financial-information/financial-information.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import {
+  UserProfileUpdateErrorResponse,
+  UserProfileUpdatedResponse
+} from 'src/app/models/Profile';
+import { Subscription } from 'rxjs';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { ProfileService } from 'src/app/services/profile/profile.service';
+import { first } from 'rxjs/operators';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-financial-information',
+  templateUrl: './financial-information.component.html',
+  styleUrls: ['financial-information.component.scss']
+})
+export class FinancialInformationComponent implements OnInit {
+  deposits: any;
+  cardInfo: any;
+  constructor(
+    private profileService: ProfileService,
+    private spinner: NgxSpinnerService
+  ) {}
+  subscribe: Subscription[] = [];
+  ngOnInit() {
+    this.getDeposits();
+    this.getCardInfo();
+    this.getCardBrand();
+    console.log('the cardInfo', this.cardInfo);
+  }
+
+  getDeposits() {
+    this.profileService
+      .getDeposits()
+      .pipe(first())
+      .subscribe(
+        response => {
+          response.data.transactions.map(transaction => {
+            if ( transaction.percentage_completion !== '100' ) {
+              this.deposits = transaction;
+            }
+            return this.deposits;
+          });
+        },
+        error => {
+          this.deposits = {error: error.error.errors};
+        }
+      );
+  }
+
+  getCardInfo() {
+    this.profileService.userProfile$.subscribe(response => {
+      const profile = response.data.profile;
+      const cardNumberFull = [...profile.card_info.card_info.card_number];
+      this.cardInfo = {
+        cardOwner : `${profile.user.first_name} ${profile.user.last_name}`,
+        cardNumber: `${cardNumberFull[9]} ${cardNumberFull[10]} ${cardNumberFull[11]} ${cardNumberFull[12]}`,
+        cardExpiry: profile.card_info.card_info.card_expiry,
+      };
+    });
+  }
+
+  getCardBrand() {
+    this.profileService.userProfile$.subscribe(response => {
```

Comment:
Hi @shemaeric  what do you think of having 1 subscription that passes the response to the `getCardInfo` and the `getCardBrand` functions. I see both functions are doing different things but they are subscribing to the same thing `this.profileService.userProfile$`

Reply:
Thank you for your suggestion @Tevinthuku. I have worked on this.